.github/workflows/check_smoke_tests.yml

```diff
@@ -41,13 +41,23 @@ jobs:
     - name: Check out repository
       uses: actions/checkout@v4
 
-    - name: Set up Docker Buildx
-      uses: docker/setup-buildx-action@v3
-
-    - name: Build RecordLinker Docker Image
-      run: |
-        docker build -t rl-service-image . 
-
+    - name: Convert repository name to lowercase
+      run: echo "REPO=$(echo '${{ github.repository }}' | tr '[:upper:]' '[:lower:]')" >> $GITHUB_ENV
+
+    - name: Log in to GitHub Container Registry
+      uses: docker/login-action@v3
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+      
+    - name: Build and push RecordLinker image
+      uses: docker/build-push-action@v3
+      with:
+          push: true
+          tags: |
+            ghcr.io/${{ env.REPO }}/rl-service:latest 
```

Comment:
I think there is a potential for a conflict here if we have more than one open PR.  If both PR's are building, the last one win's out.

Reply:
@ericbuckley I was thinking of changing the image name to be based on the pr like `rl-service:pr-$PR_NUMBER`, but want to make sure the image gets removed after the PR is merged. I could do it through a workflow or let a retention policy take care of it, but I don't know what our retention policy is and but don't seem to have the right permissions to do so. Do we have a retention policy set or should I remove the image with a workflow?

Reply:
Will make any updates in the other PR to address this

Reply:
To my knowledge we don't have a retention policy set.  I'm not even sure how to do that on a ghcr.io repo, but I'm sure there is a way.  One thing that might be worth considering is using docker layer caching for this rather than an ghcr.io repo.  Take a look at this article https://medium.com/@depot.dev/how-to-use-docker-layer-caching-in-github-actions-bee1f7b19093

Reply:
Author of ☝️ (co-founder of Depot, former PDX native), you can find a less annoying & updated version (Medium can be meh) of that post here: https://depot.dev/blog/docker-layer-caching-in-github-actions

Also, if you want to skip pushing to GHCR to use as the cache, you can specify `load: true`, and that _should_ just get the image back to you in its tagged form to then do a `docker run` on it. You could then do the `docker push` after your tests to put it in GHCR.

If you also want an open source sponsorship of Depot to swap `docker/build-push-action` for `depot/build-push-action`, also happy to set that up.

Reply:
Thanks @kylegalbraith I appreciate the guidance.